client/src/context/studyappContext.js

```diff
@@ -1,18 +1,22 @@
 import React, { createContext, useContext, useReducer } from 'react';
+import Profile from '../models/Profile';
 
 const AppContext = createContext({
     isLoading: true,
     isAuth: false,
+    profile: {},
 });
 
 const initialState = {
     isLoading: true,
+    profile: Profile,
 };
 
 const reducer = (state, action) => {
     const { type, payload } = action;
     switch (type) {
-        case 'CASE_ONE':
+        case 'updateProfile':
+            state.profile = payload;
```

Comment:
You can move profile in the returned object just below ...state. Otherwise updating a state like that might create unpredictable mutation.